src/context/color-mode/ProvideColorMode.State.ts

```diff
@@ -7,34 +7,43 @@ export interface ColorModeContext {
   toggleColorMode: () => void;
 }
 
-type Mode = "light" | "dark";
+type Mode = "light" | "dark" | "system";
 
 const useProvideColorMode = () => {
-  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)", {
-    noSsr: true,
-  })
+  const systemColorScheme = useMediaQuery("(prefers-color-scheme: dark)")
     ? "dark"
     : "light";
 
-  const [mode, setMode] = useState<Mode>(
-    (localStorage.getItem("color_scheme") as Mode) || prefersDarkMode,
-  );
+  const [mode, setMode] = useState<Mode>(() => {
+    const storedMode = localStorage.getItem("color_scheme") as Mode;
+    return storedMode === "light" || storedMode === "dark"
+      ? storedMode
+      : "system";
+  });
 
   const toggleColorMode = () => {
     setMode((prevMode) => {
       if (prevMode === "light") {
         localStorage.setItem("color_scheme", "dark");
         return "dark";
-      } else {
+      } else if (prevMode === "dark") {
         localStorage.setItem("color_scheme", "light");
         return "light";
+      } else {
+        localStorage.removeItem("color_scheme");
```

Comment:
I think there is a bug here, which I also was able to confirm via the preview.
In the initial state, that is the first time a user presses the toggle it will go into the `else` branch of the code and set the "new theme" to the current system theme (aka nothing happens). Only when pressing the light/dark toggle a second time (and during the same browser session) it will actually toggle.